File: patch_routes.js
import fs from 'fs';

let routes = fs.readFileSync('routes.ts', 'utf8');

// replace parseInt with validation
routes = routes.replace(/const id = parseInt\(req\.params\.id, 10\);/g, "const id = parseInt(req.params.id, 10);\n      if (isNaN(id)) return res.status(400).json({ error: 'Invalid ID format' });");

fs.writeFileSync('routes.ts', routes);

File: patch_routes.py
import re

with open('routes.ts', 'r') as f:
    routes = f.read()

# Replace the catch blocks in GET routes
routes = re.sub(
    r'    } catch \(e\) \{\n      next\(e\);\n    \}',
    r'''    } catch (e: any) {
      if (e?.message?.includes('429') || e?.message?.includes('Quota') || e?.code === 429) {
        console.warn('Firebase Quota Exceeded. Returning empty result.');
        res.status(429).json({ error: "Firebase Quota Exceeded. Please try again later." });
      } else {
        next(e);
      }
    }''',
    routes
)

with open('routes.ts', 'w') as f:
    f.write(routes)
